src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/Lobby.java

```diff
@@ -100,6 +107,33 @@ public String getInvitationCode() {
         return this.invitationCode;
     }
 
+    @Override
+    public void setUserName(String token, String newName) {
+        if (userNames.contains(newName))
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "Username " + newName + " is already taken.");
```

Comment:
Do we want to throw ResponseStatusExceptions in the Lobby and LobbyManager? I just ask myself if this couples them to much to the API. If these methods like setUserName get called outside of the context of a API request, the error makes no sense. Wouldn't it be better to throw something like a IlligalArgumentException (maybe there is something more appropriate) and catch them in the service?

Reply:
or maybe do the check in the lobby service and the setter has as @precondition that it must be unique

Reply:
I think we can either:
1. Use different exceptions (e.g. DuplicateUsernameException, DuplicateLobbyNameException, etc.).
2. Use exceptions with the actual error specified in REST interface to return to the client (ResponseStatusException).
As for using the ResponseStatusException outside API calls, are there any plans to run any of this code outside of the server so that entry points are not web calls?

Using something else, such as IllegalArgumentException will force us to either duplicate the checks (one in the caller, another in the callee) or group the error codes for REST into a single INTERNAL_SERVER_ERROR. If we need to have different error codes for different reasons (as stated in the REST specification), grouping them should not be a valid option.


Will leave it as is while a decision is made. Just let me know and I'll change this.